### ISSUE TITLE
[feature request] Add emmision material/shader

### ISSUE THREAD

**Issue by MisterMakerNL** [FreeCAD/FreeCAD-render#315]
So I wanted to make another lamp but a real shortcoming here is that I cannot. Do an emission shader on an object. (a surface that emits light.)
 I do think at least cycles supports it:
https://github.com/blender/cycles/blob/ccc73ccc570a92f59b0cd63f56a688a7782e346d/src/scene/shader_graph.h#L175
So would be nice if there would be an emitting material. We can also use it for screens and other electronics with leds.
![0aaf866ca8b397838e566764194a2599](https://github.com/FreeCAD/FreeCAD-render/assets/29804962/18d92354-52e6-44bd-b210-224058562b18)

**Comment by MisterMakerNL** [comment]
I realize this is exactly what the area-light does, but I kinda would like being able to apply the area light to a object like a light-bulb!

**Comment by howetuft** [comment]
Yes, that's a good idea! I've started something as a proof-of-concept. You can find it in the `emission` branch of my personal repo: https://github.com/howetuft/FreeCAD-render/tree/emission. At now, it's restricted to Cycles - other renderers still to be coded -, and it may also evolve a bit, but I would be interested by your feedback.

**Comment by howetuft** [comment]
Hello,
#323 brings emission material support: can you try, please? I'm eager to see the bulb enlightened!

**Comment by MisterMakerNL** [comment]
Yes, this is cool! 
Although my renders below is maybe a bit of a bad use case, as the light shouldn't be the object of focus.

![image](https://github.com/FreeCAD/FreeCAD-render/assets/29804962/123da123-9726-405a-a147-03eafd529cea)
Red light
![image](https://github.com/FreeCAD/FreeCAD-render/assets/29804962/0e98b9e5-6cf2-4464-b678-de4fc21666cf)

**Comment by howetuft** [comment]
Wow, amazing!

**Comment by MisterMakerNL** [comment]
Now I made one how it should be used :)
[Model credits](https://grabcad.com/library/wireless-desk-reading-lamp-1) (not a Freecad model)
![image](https://github.com/FreeCAD/FreeCAD-render/assets/29804962/a6da9c5b-e13e-424b-855a-c3d2404ea500)

**Comment by MisterMakerNL** [comment]
Just in time for christmass lights XD

**Comment by howetuft** [comment]
Really impressive! It's rendered with Cycles, isn't it?

**Comment by MisterMakerNL** [comment]
> Really impressive! It's rendered with Cycles, isn't it?

Yes, I really like how smooth the model is I imported it as an iges.
